numba/core/overload_glue.py

```diff
@@ -151,16 +150,69 @@ def body(tyctx):
                     raise errors.InternalError(msg)
                 return sig, lowering
 
-            stub = self._stub_generator(len(ol_args), body, ol_kwargs)
+            # Need a typing context now so as to get a signature and a binding
+            # for the kwarg order.
+            from numba.core.target_extension import (dispatcher_registry,
+                                                     resolve_target_str,
+                                                     current_target)
+            disp = dispatcher_registry[resolve_target_str(current_target())]
+            typing_context = disp.targetdescr.typing_context
+            typing = self._TYPER(typing_context)
+            sig = typing.apply(ol_args, ol_kwargs)
+            if not sig:
+                # No signature is a typing error, there's no match, so report it
+                raise errors.TypingError("No match")
+
+            # The following code branches based on whether the signature has a
+            # "pysig", if it does, it's from a CallableTemplate and
+            # specialisation is required based on precise arg/kwarg names and
+            # default values, if it does not, then it just requires
+            # specialisation based on the arg count.
+            #
+            # The "gen_var_names" function is defined to generate the variable
+            # names at the call site of the intrinsic.
+            #
+            # The "call_str_specific" is the list of args to the function
+            # returned by the @overload, it has to have matching arg names and
+            # kwargs names/defaults if the underlying typing template supports
+            # it (CallableTemplate), else it has to have a matching number of
+            # arguments (AbstractTemplate). The "call_str" is the list of args
+            # that will be passed to the intrinsic that deals with typing and
+            # selection of the lowering etc, so it just needs to be a list of
+            # the argument names.
+
+            if sig.pysig: # CallableTemplate, has pysig
+                pysig_params = sig.pysig.parameters
+
+                # Define the generator for the var names
+                def gen_var_names():
+                    for x in pysig_params.keys():
+                        yield x
```

Comment:
Seems overly complicated to use a generator. It can just be a sequence.

Reply:
list-comp used in 65ca6a0